### PR TITLE
[Woo POS][Cash & Receipts] Adapt send receipt modal to full screen

### LINKED ISSUE #14480: [Woo POS] [Cash & Receipts] New UI for sending receipt (full-screen)
Success case. 
- What happens while the email is being sent? Is there any loading screen/indicator to show here?
- Are we showing the user something after receipt is sent successfully ?
- Focus on the keyboard when entering the receipt form

Be sure to handle UX for multiple attempts (spinners, button disabled while loading, ... )

>  multiple taps on Send result in multiple emailed receipts. It also could do with a spinner state so that the user can tell their action is being handled.

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #14480

## Description
This PR updates the "send receipt" modal view to a full-screen view, and simplifies its design:

https://github.com/user-attachments/assets/e9981cef-7c07-4fe8-a5ad-6d936f2b3a50

## Testing
Better tested with a physical device, otherwise we can switch the built-in keyboard within the simulator, under I/O > Keyboard > Toggle keyboard

- Set feature flag `.sendReceiptsForPointOfSale` to true
- Log into Povilas testing site https://lovely-aware-sturgeon.jurassic.ninja/ details: PdfdoF-5Qj-p2
- Proceed with a POS transaction
- Tap on the "Receipt" button, observe we switch the view state to the new "send receipt" view
- Observe that the focus is set to the textfield
- Enter an email address, tap "send"
- Observe that the text is changed to a loading indicator while the receipt is being requested

Screenshot 2024-12-10 at 12 52 13

Once the email is sent, the view should be switched back to "TotalsView" automatically.

At the moment there is no email validation, or error handling, so the flow will end successfully despite entering an invalid email address or if there's a receipt error. This will be tackled separately.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/PaymentButtons.swift [hunk 1]
```diff
@@ -95,7 +87,7 @@ private extension PaymentsActionButtons {
         itemsController: PointOfSalePreviewItemsController(),
         cardPresentPaymentService: CardPresentPaymentPreviewService(),
         orderController: PointOfSalePreviewOrderController())
-    PaymentsActionButtons(isShowingReceiptNotEligibleBanner: .constant(true))
+    PaymentsActionButtons( isShowingSendReceiptView: .constant(false), isShowingReceiptNotEligibleBanner: .constant(true))
```
Comment (jaclync) [inline WooCommerce/Classes/POS/Presentation/PaymentButtons.swift]: super nit:
```suggestion
    PaymentsActionButtons(isShowingSendReceiptView: .constant(false), isShowingReceiptNotEligibleBanner: .constant(true))
```

**Review by jaclync — APPROVED** [review]
LGTM :shipit: 

One thing I noticed from testing:

Tapping outside of the "Send" text in the button didn't trigger the action:




---

And a non-blocking observation:

When the email is being sent with the spinner shown, it's still possible to tap <- and go back. In this case, do we want to show a success/failure notice on completion?

File: WooCommerce/Classes/POS/Presentation/PaymentButtons.swift (hunk 1 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/POS/Presentation/PaymentButtons.swift]: Fixed! https://github.com/woocommerce/woocommerce-ios/pull/14639/commits/8a055edadbf9be5b5ef8c1b310a0503bcbbc6fdf

**Comment by iamgabrielma** [comment]
Thanks for the review!

> Tapping outside of the "Send" text in the button didn't trigger the action:

Oops, good catch. Fixed: https://github.com/woocommerce/woocommerce-ios/pull/14639/commits/40274fa11e07d772531a9a2d11688b6cc3ba09a5

> When the email is being sent with the spinner shown, it's still possible to tap <- and go back. In this case, do we want to show a success/failure notice on completion?

Yeah, I think we can disable button interaction while is waiting for receipt response. I'll handle this with the rest of error handling here: https://github.com/woocommerce/woocommerce-ios/pull/14610 

Unfortunately we don't have any designs for error state so I'll double check with Android what are we going to show, perhaps we can reuse the "toast" we already use for non eligible sites and adapt it to show an error message if sending the receipt was not successful.

**Pull request merged**